client/src/components/UserManager/index.tsx

```diff
@@ -0,0 +1,157 @@
+import './styles.scss';
+
+import { Modal, SearchBar, UserCard, UserFields, UserForm, UserLocation } from 'components';
+import {
+  useGetPhotosQuery,
+  useGetUsersQuery,
+  User,
+  useUpdateUserMutation,
+} from 'graphql/_generated';
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+export interface SearchParams {
+  q?: string;
+  page?: string;
+}
+
+export default function UserManager() {
+  /*
+   *
+   * Data fetching / persistence
+   *
+   *
+   */
+
+  // Handle user search and pagination with query string
+  const [searchParams, setSearchParams] = useSearchParams();
+  const initialQuery = searchParams.get('q') ?? '';
+  const updateUrl = (updatedSearchParams: SearchParams): void => {
+    const current = Object.fromEntries(searchParams.entries()) ?? {};
+    const merged = { ...current, ...updatedSearchParams };
+    setSearchParams(Object.entries(merged));
+  };
+  const page = parseInt(searchParams.get('page') ?? '1');
+  const searchUsersQuery = searchParams.get('q');
+
+  // user data graphql query
+  const {
+    data: userData,
+    loading: usersLoading,
+    error: usersError,
+  } = useGetUsersQuery({
+    variables: { page, ...(searchUsersQuery ? { search: searchUsersQuery } : {}) },
+  });
+
+  const users = userData?.users?.users ?? [];
+  const count = userData?.users?.count ?? 0;
+
+  // photos graphql query
+  const { data: photoData, loading: photosLoading, error: photosError } = useGetPhotosQuery();
+  const photos = photoData?.getPhotos.map(({ url }) => url) || [];
+
+  // user update mutation
+  const [updateUserMutation] = useUpdateUserMutation();
+
+  // scroll the last user card into view in order to save users place on page reload
+  // (disable for fist page load and around filter/search)
+  const shouldScrollTo = (i: number): boolean => {
+    return !!(
+      users?.length !== 6 &&
+      i === users?.length - 1 &&
+      document.activeElement?.id !== 'userSearch'
+    );
+  };
+
+  /*
+   *
+   * Local state and user editing
+   *
+   *
+   */
+  const [currentlyEditingUserId, setCurrentlyEditingUserId] = useState<string | null>(null);
+  const [userAddress, setUserAddress] = useState<string | null>(null);
+  const currentlyEditingUser = users.find((u: User) => u.id === currentlyEditingUserId) ?? null;
```

Comment:
why the need for `?? null`;

```suggestion
  const currentlyEditingUser = users.find((u: User) => u.id === currentlyEditingUserId);
```

Reply:
Good question and I can't think of any 😁